stress/src/metrics_latency.rs

```diff
@@ -0,0 +1,339 @@
+    static CURRENT_RNG: RefCell<rngs::SmallRng> = RefCell::new(rngs::SmallRng::from_entropy());
+}
+
+// copy/paste from opentelemetry-sdk/benches/metric.rs
+#[derive(Clone, Debug)]
+pub struct SharedReader(Arc<dyn MetricReader>);
+
+impl SharedReader {
+    pub fn new<R>(reader: R) -> Self
+    where
+        R: MetricReader,
+    {
+        Self(Arc::new(reader))
+    }
+}
+
+impl MetricReader for SharedReader {
+    fn register_pipeline(&self, pipeline: Weak<Pipeline>) {
+        self.0.register_pipeline(pipeline)
+    }
+
+    fn collect(&self, rm: &mut ResourceMetrics) -> Result<(), MetricError> {
+        self.0.collect(rm)
+    }
+
+    fn force_flush(&self) -> Result<(), MetricError> {
+        self.0.force_flush()
+    }
+
+    fn shutdown(&self) -> Result<(), MetricError> {
+        self.0.shutdown()
+    }
+
+    fn temporality(&self, kind: InstrumentKind) -> Temporality {
+        self.0.temporality(kind)
+    }
+}
+
+fn main() {
+    let available_threads: usize = std::thread::available_parallelism().map_or(1, |p| p.get());
+
+    for threads_count in [available_threads / 4, available_threads * 4] {
+        println!("*** updates, using {threads_count} threads ***");
+        measure_update_latency(&format!("no attribs"), threads_count, 10000000, |_i, _j| []);
+        measure_update_latency(&format!("1 attrib"), threads_count, 10000000, |_i, _j| {
+            [KeyValue::new("some_key", 1)]
+        });
+
+        measure_update_latency(&format!("9 attribs"), threads_count, 10000000, |_i, _j| {
+            // for http.server.request.duration as defined in https://opentelemetry.io/docs/specs/semconv/http/http-metrics/
+            [
+                KeyValue::new("http.request.method", "GET"),
+                KeyValue::new("url.scheme", "not_found"),
+                KeyValue::new("error.type", 404),
+                KeyValue::new("http.response.status_code", 404),
+                KeyValue::new("http.route", "testing/metrics/latency"),
+                KeyValue::new("network.protocol.name", "http"),
+                KeyValue::new("network.protocol.version", 2),
+                KeyValue::new("server.address", "example.com"),
+                KeyValue::new("server.port", 8080),
+            ]
+        });
+        println!("*** inserts, using {threads_count} threads ***");
+        measure_update_latency(&format!("1 attrib"), threads_count, 1500, |i, j| {
+            [KeyValue::new(format!("some_key{i}"), j as i64)]
+        });
+
+        measure_update_latency(&format!("10 attribs"), threads_count, 1500, |i, j| {
+            [
+                KeyValue::new(format!("random{i}"), j as i64),
+                KeyValue::new("http.request.method", "GET"),
+                KeyValue::new("url.scheme", "not_found"),
+                KeyValue::new("error.type", 404),
+                KeyValue::new("http.response.status_code", 404),
+                KeyValue::new("http.route", "testing/metrics/latency"),
```

Comment:
one interesting thing is - the route may not be readily available as a `static str always, forcing users to String allocate to report this!
https://github.com/open-telemetry/opentelemetry-rust/pull/2089#issuecomment-2341751172 is trying to fix that scenario, hopefully we can make it happen for 1.0 stable of Metrics!